pkg/machines/getOSList.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/python3
+
+import gi
+gi.require_version('Libosinfo', '1.0')
+from gi.repository import Libosinfo
+import sys
+
+
+loader = Libosinfo.Loader()
+loader.process_default_path()
+db = loader.get_db()
+
+oses = db.get_os_list()
+for i in range(oses.get_length()):
+    os = oses.get_nth(i)
+
+    osId = os.get_id()
+    if not osId:
+        osId = ""
```

Comment:
Please use the pattern `varname = call() or ""` to avoid having to repeat this pattern so often.